pkg/controller/direct/compute/forwardingrule_controller.go

```diff
@@ -379,11 +379,38 @@ func (a *forwardingRuleAdapter) Update(ctx context.Context, updateOp *directbase
 	forwardingRule.Name = direct.LazyPtr(a.id.forwardingRule)
 	forwardingRule.Labels = desired.Labels
 
-	// Patch only support update on networkTier field, which KCC does not support yet.
-	// Use setTarget and setLabels to update target and labels fields.
 	var err error
 	op := &gcp.Operation{}
 	updated := &computepb.ForwardingRule{}
+	if !reflect.DeepEqual(forwardingRule.AllowGlobalAccess, a.actual.AllowGlobalAccess) {
+		// To match the request body in TF-controller log
+		// https://github.com/hashicorp/terraform-provider-google/blob/main/google/services/compute/resource_compute_forwarding_rule.go#L1151
+		reqBody := &computepb.ForwardingRule{AllowGlobalAccess: forwardingRule.AllowGlobalAccess}
+		if a.id.location == "global" {
+			// TF does not support allowGlobalAccess field for global forwarding rule
+			// Underlying API as well, error message: `Field allow-global-access is only supported for regional INTERNAL
+			// forwarding rules with backend service/target instance or regional INTERNAL_MANAGED forwarding rules.`
+			forwardingRule.AllowGlobalAccess = nil
```

Comment:
@yuwenma similar to your previous [comment](https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/2724#discussion_r1767566686) for networkTier, I guess we should NOT drop this field for global forwarding rule, send the request as is and let the API to return an error if the field is set accidentally by user?